subprojects/plugins/src/main/java/org/gradle/api/tasks/SourceSet.java

```diff
@@ -25,10 +25,13 @@
 import javax.annotation.Nullable;
 
 /**
- * A {@code SourceSet} represents a logical group of Java source and resources.
+ * A {@code SourceSet} represents a logical group of Java source and resource files. They
+ * are covered in more detail in the
+ * <a href="/userguide/building_java_projects.html#sec:java_source_sets">user manual</a>.
```

Comment:
we should try to use relative links, if possible, rather than assume a particular layout. This way, for example, links in the Javadoc in the `-all` distribution will work, and links in Javadocs for versions other than the most recent Gradle version will point to the correct version of the user manual.

Reply:
The version is handled by Javascript on the docs.gradle.org site, so the above should work for '/current/', '/release-candidate/', '/5.0/' and so on. I hadn't considered the implication for 'file://' URLs, but the relative URL worried me because the relative path is different depending on whether the class is open in the HTML Javadocs or the Groovy DSL Reference.

The only "reliable" solution that I can see is to use a fully qualified URL, but that means hard-coding it to a specific version of the docs online. That's what has been done elsewhere in the Javadocs.

I'm open to any suggestions that will work for all cases.

Reply:
A solution would be to choose a contract for links from javadoc to the user manual and use this contract everywhere in the Javadocs. The tooling would rework these links as necessary for whatever context they're building the docs for. The tooling should also fail if the contract isn't used.

Two options for links would be relative ('userguide/whatever.html') or server absolute ('/userguide/whatever.html'). When building the Javadoc and DSL guide, we can turn these into relative links.